### PR TITLE
[PyTorch] Add test for all-masked case for native softmax

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

It returns all NaNs. CUDA implementation required a fix for this.

Differential Revision: [D35327730](https://our.internmc.facebook.com/intern/diff/D35327730/)

### CODE REVIEW

**Comment by swolchok** [comment]
having issues with tooling, spammed multiple PRs
